src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusResource.cs

```diff
@@ -11,28 +12,46 @@ namespace Aspire.Hosting.Azure;
 /// <param name="name">The name of the resource.</param>
 /// <param name="configureInfrastructure">Callback to configure the Azure Service Bus resource.</param>
 public class AzureServiceBusResource(string name, Action<AzureResourceInfrastructure> configureInfrastructure)
-    : AzureProvisioningResource(name, configureInfrastructure), IResourceWithConnectionString, IResourceWithAzureFunctionsConfig
+    : AzureProvisioningResource(name, configureInfrastructure), IResourceWithConnectionString, IResourceWithAzureFunctionsConfig, IResourceWithEndpoints
 {
-    internal List<string> Queues { get; } = [];
-    internal List<string> Topics { get; } = [];
-    internal List<(string TopicName, string Name)> Subscriptions { get; } = [];
+    internal List<ServiceBusQueue> Queues { get; } = [];
+    internal List<ServiceBusTopic> Topics { get; } = [];
 
     /// <summary>
     /// Gets the "serviceBusEndpoint" output reference from the bicep template for the Azure Storage resource.
     /// </summary>
     public BicepOutputReference ServiceBusEndpoint => new("serviceBusEndpoint", this);
 
+    internal EndpointReference EmulatorEndpoint => new(this, "emulator");
+
+    /// <summary>
+    /// Gets a value indicating whether the Azure Service Bus resource is running in the local emulator.
+    /// </summary>
+    public bool IsEmulator => this.IsContainer();
+
     /// <summary>
     /// Gets the connection string template for the manifest for the Azure Service Bus endpoint.
     /// </summary>
     public ReferenceExpression ConnectionStringExpression =>
-        ReferenceExpression.Create($"{ServiceBusEndpoint}");
+        IsEmulator
+        ? ReferenceExpression.Create($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;")
+        : ReferenceExpression.Create($"{ServiceBusEndpoint}");
 
     void IResourceWithAzureFunctionsConfig.ApplyAzureFunctionsConfiguration(IDictionary<string, object> target, string connectionName)
     {
-        // Injected to support Azure Functions listener initialization.
-        target[$"{connectionName}__fullyQualifiedNamespace"] = ServiceBusEndpoint;
-        // Injected to support Aspire client integration for Service Bus in Azure Functions projects.
-        target[$"Aspire__Azure__Messaging__ServiceBus__{connectionName}__FullyQualifiedNamespace"] = ServiceBusEndpoint;
+        if (IsEmulator)
+        {
+            // Injected to support Azure Functions listener initialization.
+            target[$"{connectionName}__fullyQualifiedNamespace"] = ConnectionStringExpression;
```

Comment:
I think this is supposed to be

```suggestion
            target[$"{connectionName}__ConnectionString"] = ConnectionStringExpression;
```

Can you test this to ensure it works? Maybe add the RunAsEmulator to the Functions playground app, and remove all these skip #ifs:

https://github.com/dotnet/aspire/blob/9d8b13880db667e30a91cd77a9aa8b614420a937/playground/AzureFunctionsEndToEnd/AzureFunctionsEndToEnd.AppHost/Program.cs#L23-L25

Reply:
I am getting 

```
[2025-01-04T00:31:09.280Z] The listener for function 'Functions.MyServiceBusTrigger' was unable to start. Azure.Messaging.ServiceBus: The value 'Endpoint=sb://localhost:51234;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;' is not a well-formed Service Bus fully qualified namespace. (Parameter 'fullyQualifiedNamespace').
```

Is it possible the functions runtime is not expecting the local url for SB? Hopefully it's just a bug in this PR, but it looks like functions is correctly seeing the connection string.

Reply:
All working